x/mongo/driver/topology/rtt_monitor.go

```diff
@@ -104,37 +127,45 @@ func (r *rttMonitor) start() {
 	}
 }
 
-// runHello runs a "hello" operation using the provided connection, measures the duration, and adds
-// the duration as an RTT sample and returns the connection used. If the provided connection is nil
-// or closed, runHello tries to establish a new connection. If the "hello" operation returns an
-// error, runHello closes the connection.
-func (r *rttMonitor) runHello(conn *connection) *connection {
-	if conn == nil || conn.closed() {
-		conn := r.cfg.createConnectionFn()
+// runHellos runs "hello" operations in a loop using the provided connection, measuring and
+// recording the operation durations as RTT samples. If it encounters any errors, it returns.
+func (r *rttMonitor) runHellos(conn *connection) {
+	ticker := time.NewTicker(r.cfg.interval)
+	defer ticker.Stop()
 
-		err := conn.connect(r.ctx)
-		if err != nil {
-			return nil
+	for {
```

Comment:
To check my understanding: does `runHellos` run as many `hello`s as possible within the `timeout` and add a new sample for each successful `hello`? What's the reason for this design over the original `runHello`?